common/test/acceptance/tests/lms/test_lms_instructor_dashboard.py

```diff
@@ -444,3 +447,50 @@ def test_pending_tasks_section_is_visible(self):
             Then I see 'Pending Instructor Tasks' section
         """
         self.assertTrue(self.certificates_section.pending_tasks_section.visible)
+
+
+class DjangoSudoThirdPartyAuthTest(BaseInstructorDashboardTest):
+    """
+    Tests for third party auth on django sudo page.
+    """
+
+    def setUp(self):
+        """
+        Install a course with no content using a fixture.
+        """
+        super(DjangoSudoThirdPartyAuthTest, self).setUp()
+        self.course_fixture = CourseFixture(**self.course_info).install()
+        self.login_page = CombinedLoginAndRegisterPage(
+            self.browser,
+            start_page="login",
+            course_id=self.course_id
+        )
+        self.log_in_as_instructor()
+        self.instructor_dashboard_page = InstructorDashboardPage(self.browser, self.course_id)
+
+    def test_third_party_auth_on_sudo_page_with_unlinked(self):
+        """
+        Test that dummy auth button is disabled on sudo page when no account is linked.
+        """
+        sudo_password_page = SudoPage(self.browser, self.instructor_dashboard_page)
+        sudo_password_page.visit()
+        self.assertTrue(sudo_password_page.is_dummy_auth_button_disabled)
+
+    def test_third_party_auth_on_sudo_page_with_linked(self):
+        """
+        Test that user can authenticate on sudo page with dummy third party auth.
+        """
+        self._link_dummy_account()
+        sudo_password_page = SudoPage(self.browser, self.instructor_dashboard_page)
+        sudo_password_page.visit()
+        sudo_password_page.click_third_party_dummy_provider_button()
+        self.instructor_dashboard_page.visit()
```

Comment:
@dan-f Actually don't need to manually call visit on page because when we are clicking on dummy provider button it's actually already waiting for that page. So we don't need to assert and visit both.